sklearn/inspection/partial_dependence.py

```diff
@@ -550,21 +574,19 @@ def convert_feature(fx):
 
     features = tmp_features
 
-    names = []
-    try:
-        for fxs in features:
-            names_ = []
-            # explicit loop so "i" is bound for exception below
-            for i in fxs:
-                names_.append(feature_names[i])
-            names.append(names_)
-    except IndexError:
-        raise ValueError('All entries of features must be less than '
-                         'len(feature_names) = {0}, got {1}.'
-                         .format(len(feature_names), i))
+    if isinstance(ax, list):
```

Comment:
I did not look at the coverage yet. We need to make sure to have a test raising this one and the same check within the object.

Reply:
This is checked in `test_plot_partial_dependence_incorrent_num_axes`.